skyportal/models.py

```diff
@@ -3120,6 +3126,327 @@ class SourceNotification(Base):
     level = sa.Column(sa.String(), nullable=False)
 
 
+class GcnEvent(Base):
+    """Event information, including an event ID, mission, and time of the event."""
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+
+    dateobs = sa.Column(sa.DateTime, doc='Event time', unique=True, nullable=False)
```

Comment:
Correct me if I'm wrong, but the `dateobs` is basically the identifier for an event and you don't really use the autoincrementing ID anywhere right? If so, it might make more sense to just make `dateobs` the primary key since that's what you use in all the relationship and api logic anyways.

Reply:
yes good point.

Reply:
However, without an id column, I get: {"status": "error", "message": "(psycopg2.errors.NotNullViolation) null value in column \"id\" of relation \"gcnevents\" violates not-null constraint\nDETAIL:  Failing row contains (null, 2021-06-11 02:25:22.42374, 2021-06-11 02:25:22.42374, 2018-01-16 00:36:53).\n\n[SQL: INSERT INTO gcnevents (created_at, modified, dateobs) VALUES (timezone(%(timezone_1)s, CURRENT_TIMESTAMP), timezone(%(timezone_1)s, CURRENT_TIMESTAMP), %(dateobs)s)]\n[parameters: {'timezone_1': 'UTC', 'dateobs': datetime.datetime(2018, 1, 16, 0, 36, 53)}]\n(Background on this error at: http://sqlalche.me/e/13/gkpj)", "data": {}, "version": "0.9.dev0+git20210529.1bf42c0"}